components/application-gateway/cmd/applicationgateway/options.go

```diff
@@ -29,28 +32,33 @@ func parseArgs() *options {
 	proxyTimeout := flag.Int("proxyTimeout", 10, "Timeout for proxy call.")
 	requestLogging := flag.Bool("requestLogging", false, "Flag for logging incoming requests.")
 	proxyCacheTTL := flag.Int("proxyCacheTTL", 120, "TTL, in seconds, for proxy cache of Remote API information")
+	annotatePassportHeaders := flag.Bool("annotatePassportHeaders", false, "Flag to indicate if we should annotate passport headers to outgoing request")
```

Comment:
I'm not convinced that Application Gateway is the right place to add this functionality. Application Gateway is designed to be a generic component that is responsible for authenticating requests. We should strive to keep it simple and not litter the design with specific use cases.

I would consider the following solutions instead:

- Using Service Catalog Bindings and implementing header annotation in Serverless Function. Service Catalog offers Redis binding. You could implement a Server-less Function that will fetch  key value from database and add headers before calling Application Gateway. 

- Extending Application Gateway with generic mechanism for enriching requests.There is already #6346 issue that proposes such a mechanism.  Adding passport headers is a specific case that could be implemented with the mentioned mechanism.